### DISCUSSION TITLE
How to include entire applications (i.e apt-get packages) in my app?

### DISCUSSION THREAD

**Discussion by flancast90** [pyinstaller/pyinstaller#6633] [Help]
As the name implies, I'd like to have all the necessary installs bundled inside my project root with the executable. From looking online, all I've found is how to include specific python libs. In saying this, I don't mean a pip package, but an apt package.

Specifically, I'd like to include TShark (``sudo apt install tshark``), which requires JDK 17. Is there a built-in method for doing this?

**Comment by rokm** [comment]
> Is there a built-in method for doing this?

No.

**Reply by flancast90** [reply]
Thanks for the quick response!
Since there's no "easy" method for this, could you perhaps point me in the direction to do something like this? I can't imagine that there's no way to do this, since so many people use pyinstaller for Selenium apps, meaning that they often have to bundle a portable version of Chrome.

**Reply by rokm** [reply]
Do they really bundle portable Chrome, or do they use system-installed browser? Even so, I suppose (manually) bundling portable Chrome might be less of an issue due to it being portable, as opposed to an arbitrary system-installed package with its system-installed dependencies (and possibly distribution-specific patched behavior and search paths)...

**Reply by flancast90** [reply]
I think it's probably different for everyone, since some rely on Chrome-specific options in their app setup. I just found https://github.com/shivamgly/Google-Chrome-Portable-maker-for-linux for making Chrome portable which seems to be promising. It's just a matter of writing a shell script for linux to grab the latest .deb.

However, still no solution for Tshark/WireShark. Only portable version I found there was for Windows only.

**Reply by rokm** [reply]
I'd say that this is a very specific case, because the script is unpacking google-provided Chrome .deb package, so it's probably already pretty much portable by itself (compared to, for example, distribution-packaged Chrome).

**Reply by flancast90** [reply]
UPDATE: I've managed to get Chrome working in Linux! I copied ``opt/google/chrome`` over to my project root, and then in the code specified the binary path with ``chrome_options.set_binary('chrome/chrome')``. This way,  I can include the chrome directory when packaging with Pyinstaller. 

I'm assuming Wireshark will be similar if most of its contents are in a single directory, but testing for now.

**Reply by rokm** [reply]
> I'm assuming Wireshark will be similar if most of its contents are in a single directory

I wish I could go back to time when I had this sort of optimism... :)

On the other hand, if it does work, all the better.

**Reply by flancast90** [reply]
haha nothing compares to the feeling of solving a problem that you've been stumped on for days.
